src/main/java/core/basesyntax/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
```

Comment:
The validator is not initialized. You should pass an instance of `FruitTransactionValidatorImpl` to the `SupplyOperationHandler`, preferably through the constructor, to adhere to the Dependency Inversion principle.

Reply:
not fixed